### PR TITLE
[3625][ADD] product_group_product_manager

### PR DESCRIPTION
**Pull request by AungKoKoLin1997**
[3625](https://www.quartile.co/web#id=3625&cids=3&menu_id=505&action=1457&model=project.task&view_type=form)

### CODE REVIEW

**Review by yostashiro — CHANGES_REQUESTED** [review]
Please split the module into two:

- product_group_manage_product
- account_security_adj

File: product_group_manage_product/security/product_group_manage_product_security.xml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    </record>
+    <record id="rule_product_create_edit_delete" model="ir.rule">
+        <field name="name">Product Create/Edit/Delete Rule</field>
+        <field name="model_id" ref="product.model_product_product"/>
+        <field name="groups" eval="[(4, ref('product_group_manage_product.group_manage_product'))]"/>
+        <field name="perm_read" eval="True"/>
+        <field name="perm_write" eval="True"/>
+        <field name="perm_create" eval="True"/>
+        <field name="perm_unlink" eval="True"/>
+    </record>
```
Comment (yostashiro) [inline product_group_manage_product/security/product_group_manage_product_security.xml]: Why do we need these record rules?

File: product_group_manage_product/security/ir.model.access.csv [hunk 2]
```diff
@@ -0,0 +1,3 @@
+id,name,model_id:id,group_id:id,perm_read,perm_write,perm_create,perm_unlink
+access_product_template,access_product_template,product.model_product_template,group_manage_product,1,1,1,0
+access_product_product,access_product_product,product.model_product_product,group_manage_product,1,1,1,0
```
Comment (yostashiro) [inline product_group_manage_product/security/ir.model.access.csv]: Any reason why `perm_unlink` should be false?
```suggestion
access_product_template_manage_product,product.template.manage.product,product.model_product_template,group_manage_product,1,1,1,1
access_product_product_manage_product,product.product.manage.product,product.model_product_product,group_manage_product,1,1,1,1
```

File: product_group_manage_product/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# Copyright 2023 Quartile Limited
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl).
+{
+    "name": "Product Group Manage Product",
+    "version": "11.0.1.0.0",
+    "author": "Quartile Limited",
+    "website": "https://www.quartile.co",
+    "category": "Product",
+    "license": "LGPL-3",
```
Comment (yostashiro) [inline product_group_manage_product/__manifest__.py]:
```suggestion
    "license": "AGPL-3",
```

**Review by yostashiro — CHANGES_REQUESTED** [review]

File: product_group_manage_product/security/product_group_manage_product_security.xml [hunk 4]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+    <record id="group_manage_product" model="res.groups">
+        <field name="name">Product Creation, Editing &amp; Deletion</field>
```
Comment (yostashiro) [inline product_group_manage_product/security/product_group_manage_product_security.xml]:
```suggestion
        <field name="name">Product Manager</field>
```

**Review by yostashiro — COMMENTED** [review]
I've come to think `product_group_product_manager` might be a better choice for the module name (upon changing the technical name of the module).  What do you think?

**Comment by AungKoKoLin1997** [comment]
> I've come to think `product_group_product_manager` might be a better choice for the module name (upon changing the technical name of the module). What do you think?

Yes. `product_group_product_manager` is better module name and can easily know this module is added new group `group_product_manager`.

File: product_group_product_manager/readme/DESCRIPTION.rst [hunk 5]
```diff
@@ -0,0 +1 @@
+This module adds a new group (group_manage_product) with permissions for creating, editing, and deleting products.
```
Comment (yostashiro) [inline product_group_product_manager/readme/DESCRIPTION.rst]:
```suggestion
This module adds a new group (group_product_manager) with permissions for creating, editing, and deleting products.
```

**Review by yostashiro — APPROVED** [review]
Code review.  Thanks Lin!

**Review by kanda999 — CHANGES_REQUESTED** [review]
@AungKoKoLin1997 
I have found that if product manager don't have access to the product.price.history model, they get an error when they try to create a new product from sale.order.line or purchase.order.line.

So can product_manager please allow read write update delete of product.price.history?

**Review by kanda999 — APPROVED** [review]
Function, code review

**Pull request merged**